test/e2e/framework/framework.go

```diff
@@ -348,7 +347,7 @@ func (f *Framework) AfterEach() {
 					Logf(summaries[i].PrintHumanReadable())
 				} else {
 					// TODO: learn to extract test name and append it to the kind instead of timestamp.
-					filePath := path.Join(TestContext.ReportDir, summaries[i].SummaryKind()+now.Format(time.RFC3339)+".txt")
+					filePath := path.Join(TestContext.ReportDir, summaries[i].SummaryKind()+"_"+f.BaseName+".txt")
```

Comment:
Changed it to add timestamp too. Should work for now, but this way of identifying tests doesn't seem to be the best way as it makes life tougher for automated parsers.
How about adding the name of the test inside the file itself as the first line (as it seems too long to be appended to the filename)? This it's a bit hacky as we have to remove the first line before parsing json.